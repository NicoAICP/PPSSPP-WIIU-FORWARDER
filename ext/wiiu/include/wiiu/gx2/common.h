#pragma once

#define GX2_SCAN_BUFFER_ALIGNMENT      0x1000
#define GX2_SHADER_ALIGNMENT           0x100
#define GX2_CONTEXT_STATE_ALIGNMENT    0x100
#define GX2_UNIFORM_BLOCK_ALIGNMENT    0x100
#define GX2_DISPLAY_LIST_ALIGNMENT     0x20
#define GX2_VERTEX_BUFFER_ALIGNMENT    0x40
#define GX2_INDEX_BUFFER_ALIGNMENT     0x20

#define GX2_ENABLE                     TRUE
#define GX2_DISABLE                    FALSE

#define GX2_TRUE                       TRUE
#define GX2_FALSE                      FALSE

#ifndef GX2_COMP_SEL
#define GX2_COMP_SEL(c0, c1, c2, c3) (((c0) << 24) | ((c1) << 16) | ((c2) << 8) | (c3))
#ifdef __cplusplus
#include <functional>
#endif
#include <stdio.h>

#define _x 0
#define _y 1
#define _z 2
#define _w 3
#define _r 0
#define _g 1
#define _b 2
#define _a 3
#define _0 4
#define _1 5
#endif


#define GX2_DISABLE_WRAPS
