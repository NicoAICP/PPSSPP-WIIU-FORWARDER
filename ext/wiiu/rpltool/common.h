#ifndef COMMON_H
#define COMMON_H

#include <ctype.h>
#include <stdint.h>
#include <stdio.h>

typedef int8_t s8;
typedef int16_t s16;
typedef int32_t s32;

typedef uint8_t u8;
typedef uint16_t u16;
typedef uint32_t u32;

#define GET_ENUM_NAME(prefix, name, val) \
   case prefix##name: \
      return #name;

#define EMIT_ENUM_TO_STR_FUNC(name, cases) \
   static inline const char *name##_to_str(name type) { \
      switch (type) { cases } \
      return "unknown"; \
   }

#define GET_FLAG_NAME(prefix, name, val) \
   case prefix##name: \
      pos += snprintf(buffer + pos, sizeof(buffer) - pos, "%s%s", #name, seperator); \
      flags &= ~(1u << i); \
      break;

#define EMIT_FLAGS_TO_STR_FUNC(name, cases) \
   static inline const char *name##_to_str(int flags, const char *seperator) { \
      int i; \
      static char buffer[32 * 32]; \
      int pos = 0; \
      for (i = 0; i < 32; i++) \
         if ((1u << i) & flags) \
            switch (1u << i) { cases } \
      if (flags) \
         snprintf(buffer + pos, sizeof(buffer) - pos, "0x%08X", flags); \
      else if (pos) \
         buffer[pos - strlen(seperator)] = '\0'; \
      else \
         return "NULL"; \
      return buffer; \
   }

#define MAKE_MAGIC(a, b, c, d) (((a) << 24) | ((b) << 16) | ((c) << 8) | ((d) << 0))

#define log_var(X) printf("%-35s: 0x%0*X (%u)\n", #X, sizeof(X) * 2, (u32)(X), (u32)(X))
#define log_svar(X) printf("%-35s: 0x%0*X (%i)\n", #X, sizeof(X) * 2, (u32)(X), (s32)(X))
#define log_array(X) debug_array(X, sizeof(X), #X)
#define log_enum(type, X) printf("%-35s: 0x%X (%s)\n", #X, (X), type##_to_str(X))
#define log_flags(type, X) printf("%-35s: 0x%08X (%s)\n", #X, (X), type##_to_str(X, " | "))
#define log_magic(X) \
   printf("%-35s: '%c''%c''%c''%c' (0x%08X)\n", #X, (u32)(X) >> 24, (u32)(X) >> 16, (u32)(X) >> 8, (u32)(X), (u32)(X))
#define log_carray(X) debug_char_array(X, sizeof(X), #X)
#define log_str(X) printf("%-35s: \"%s\"\n", #X, (char *)(X))
#define log_off_str(X, data) printf("%-35s: 0x%08X \"%s\"\n", #X, (X), (char *)data + (X))
#define log_off_strs(X, data) \
   do { \
      printf("%-35s: 0x%08X ", #X, (X)); \
      const char *ptr = (char *)data + (X); \
      while (*ptr) { \
         ptr += printf("\"%s\" ", ptr) - 2; \
      } \
      printf("\n"); \
   } while (0)

static inline void dump_array(uint32_t *src, int size);
static inline void debug_array(uint32_t *src, int size, const char *name) {
   printf("%-35s:", name);
   dump_array(src, size);
}
static inline void debug_char_array(uint8_t *src, int size, const char *name) {
   int i;
   printf("%-35s: ", name);
   //   for (i = 0; i < size; i++)
   //      printf("%-2c ", isalnum(src[i])?src[i]: '?');
   //   printf("\n%-35s: ", "");
   for (i = 0; i < size; i++)
      printf("%02X ", src[i]);
   printf("\n");
}

static inline void dump_array(uint32_t *src, int size) {
   int i;
   size = (size / 4) - 1;
   printf("\n{");
   for (i = 0; i < size; i++) {
      if (!(i & 3))
         printf("\n   ");

      printf("0x%08X, ", src[i]);
   }

   if (!(i & 3))
      printf("\n   ");

   printf("0x%08X\n}\n", src[i]);
}

static inline u32 align_up(u32 x, u32 align) { return (x + (align - 1)) & ~(align - 1); }
static inline u32 align_down(u32 x, u32 align) { return x & ~(align - 1); }

#endif // COMMON_H
