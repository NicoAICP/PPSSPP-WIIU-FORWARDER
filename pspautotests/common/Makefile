TARGET_LIB = libcommon.a
EXTRA_TARGETS = libcommon_kernel.a
OBJS = common.o vram.o sysmem-imports.o
KERNEL_OBJS = common_kernel.o vram.o sysmem-imports.o

INCDIR = 
CFLAGS = -O0 -G 0 -g -Wall
CXXFLAGS = $(CFLAGS) -fno-exceptions -fno-rtti
ASFLAGS = $(CFLAGS)

LIBDIR =
LDFLAGS = -G0

PSPSDK=$(shell psp-config --pspsdk-path)
include $(PSPSDK)/lib/build.mak

%.o: %.S
	$(AS) $(ASFLAGS) -c -o $@ $<

common_kernel.o: common.c
	$(CC) $(CFLAGS) -DCOMMON_KERNEL=1 -c -o $@ $<

libcommon_kernel.a: $(KERNEL_OBJS)
	$(AR) cru $@ $(KERNEL_OBJS)
	$(RANLIB) $@
