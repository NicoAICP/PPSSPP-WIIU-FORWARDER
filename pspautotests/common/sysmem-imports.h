extern int sceKernelSetCompiledSdkVersion(int ver);
extern int sceKernelSetCompiledSdkVersion370(int ver);
extern int sceKernelSetCompiledSdkVersion380_390(int ver);
extern int sceKernelSetCompiledSdkVersion395(int ver);
extern int sceKernelSetCompiledSdkVersion401_402(int ver);
extern int sceKernelSetCompiledSdkVersion500_505(int ver);
extern int sceKernelSetCompiledSdkVersion507(int ver);
extern int sceKernelSetCompiledSdkVersion600_602(int ver);
extern int sceKernelSetCompiledSdkVersion603_605(int ver);
extern int sceKernelSetCompiledSdkVersion606(int ver);
extern int sceKernelGetCompiledSdkVersion();