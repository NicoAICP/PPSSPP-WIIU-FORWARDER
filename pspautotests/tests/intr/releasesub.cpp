#include <common.h>
#include <pspintrman.h>

void handler(int no, int *arg) {
	checkpoint("* handler(%08x, %08x)", no, arg);
}

extern "C" int main(int argc, char *argv[]) {
	int result;

	checkpointNext("sceKernelReleaseSubIntrHandler interrupts:");
	for (int intr = -2; intr < 70; ++intr) {
		sceKernelRegisterSubIntrHandler(intr, 0, (void *)&handler, (void *)0xDEADBEEF);
		result = sceKernelReleaseSubIntrHandler(intr, 0);

		checkpoint("  Interrupt %d: %08x", intr, result);
	}

	checkpointNext("sceKernelReleaseSubIntrHandler sub interrupts:");
	for (int intr = -2; intr < 70; ++intr) {
		sceKernelRegisterSubIntrHandler(30, intr, (void *)&handler, (void *)0xDEADBEEF);
		result = sceKernelReleaseSubIntrHandler(30, intr);

		checkpoint("  Sub interrupt %d: %08x", intr, result);
	}

	checkpointNext("sceKernelReleaseSubIntrHandler handlers:");
	sceKernelRegisterSubIntrHandler(30, 1, NULL, (void *)0xDEADBEEF);
	result = sceKernelReleaseSubIntrHandler(30, 1);
	checkpoint("  NULL handler: %08x", result);
	result = sceKernelReleaseSubIntrHandler(30, 1);
	checkpoint("  Twice NULL: %08x", result);
	sceKernelRegisterSubIntrHandler(30, 1, (void *)&handler, (void *)0xDEADBEEF);
	result = sceKernelReleaseSubIntrHandler(30, 1);
	checkpoint("  Not NULL: %08x", result);
	result = sceKernelReleaseSubIntrHandler(30, 1);
	checkpoint("  Twice not NULL: %08x", result);

	return 0;
}