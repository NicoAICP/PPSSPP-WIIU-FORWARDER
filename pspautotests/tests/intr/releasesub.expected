[x] sceKernelReleaseSubIntrHandler interrupts:
[x]   Interrupt -2: 80020065
[x]   Interrupt -1: 80020065
[x]   Interrupt 0: 80020068
[x]   Interrupt 1: 80020068
[x]   Interrupt 2: 80020068
[x]   Interrupt 3: 80020068
[x]   Interrupt 4: 80020068
[x]   Interrupt 5: 80020068
[x]   Interrupt 6: 80020068
[x]   Interrupt 7: 80020065
[x]   Interrupt 8: 80020068
[x]   Interrupt 9: 80020068
[x]   Interrupt 10: 80020065
[x]   Interrupt 11: 80020068
[x]   Interrupt 12: 80020065
[x]   Interrupt 13: 80020068
[x]   Interrupt 14: 80020068
[x]   Interrupt 15: 80020065
[x]   Interrupt 16: 80020065
[x]   Interrupt 17: 80020065
[x]   Interrupt 18: 80020065
[x]   Interrupt 19: 80020065
[x]   Interrupt 20: 80020065
[x]   Interrupt 21: 80020068
[x]   Interrupt 22: 80020065
[x]   Interrupt 23: 80020065
[x]   Interrupt 24: 80020065
[x]   Interrupt 25: 00000000
[x]   Interrupt 26: 80020065
[x]   Interrupt 27: 80020068
[x]   Interrupt 28: 80020068
[x]   Interrupt 29: 80020068
[x]   Interrupt 30: 00000000
[x]   Interrupt 31: 80020065
[x]   Interrupt 32: 80020068
[x]   Interrupt 33: 80020068
[x]   Interrupt 34: 80020068
[x]   Interrupt 35: 80020068
[x]   Interrupt 36: 80020065
[x]   Interrupt 37: 80020068
[x]   Interrupt 38: 80020068
[x]   Interrupt 39: 80020068
[x]   Interrupt 40: 80020068
[x]   Interrupt 41: 80020068
[x]   Interrupt 42: 80020068
[x]   Interrupt 43: 80020068
[x]   Interrupt 44: 80020068
[x]   Interrupt 45: 80020068
[x]   Interrupt 46: 80020068
[x]   Interrupt 47: 80020068
[x]   Interrupt 48: 80020068
[x]   Interrupt 49: 80020068
[x]   Interrupt 50: 80020065
[x]   Interrupt 51: 80020068
[x]   Interrupt 52: 80020068
[x]   Interrupt 53: 80020068
[x]   Interrupt 54: 80020068
[x]   Interrupt 55: 80020068
[x]   Interrupt 56: 80020065
[x]   Interrupt 57: 80020065
[x]   Interrupt 58: 80020065
[x]   Interrupt 59: 80020065
[x]   Interrupt 60: 80020065
[x]   Interrupt 61: 80020065
[x]   Interrupt 62: 80020068
[x]   Interrupt 63: 80020068
[x]   Interrupt 64: 80020068
[x]   Interrupt 65: 80020065
[x]   Interrupt 66: 80020068
[x]   Interrupt 67: 80020065
[x]   Interrupt 68: 80020065
[x]   Interrupt 69: 80020065

[x] sceKernelReleaseSubIntrHandler sub interrupts:
[x]   Sub interrupt -2: 80020065
[x]   Sub interrupt -1: 80020065
[x]   Sub interrupt 0: 00000000
[x]   Sub interrupt 1: 00000000
[x]   Sub interrupt 2: 00000000
[x]   Sub interrupt 3: 00000000
[x]   Sub interrupt 4: 00000000
[x]   Sub interrupt 5: 00000000
[x]   Sub interrupt 6: 00000000
[x]   Sub interrupt 7: 00000000
[x]   Sub interrupt 8: 00000000
[x]   Sub interrupt 9: 00000000
[x]   Sub interrupt 10: 00000000
[x]   Sub interrupt 11: 00000000
[x]   Sub interrupt 12: 00000000
[x]   Sub interrupt 13: 00000000
[x]   Sub interrupt 14: 00000000
[x]   Sub interrupt 15: 00000000
[x]   Sub interrupt 16: 80020068
[x]   Sub interrupt 17: 80020068
[x]   Sub interrupt 18: 80020068
[x]   Sub interrupt 19: 80020068
[x]   Sub interrupt 20: 80020068
[x]   Sub interrupt 21: 80020068
[x]   Sub interrupt 22: 80020068
[x]   Sub interrupt 23: 80020068
[x]   Sub interrupt 24: 80020068
[x]   Sub interrupt 25: 80020068
[x]   Sub interrupt 26: 80020068
[x]   Sub interrupt 27: 80020068
[x]   Sub interrupt 28: 80020068
[x]   Sub interrupt 29: 80020068
[x]   Sub interrupt 30: 80020068
[x]   Sub interrupt 31: 80020068
[x]   Sub interrupt 32: 80020065
[x]   Sub interrupt 33: 80020065
[x]   Sub interrupt 34: 80020065
[x]   Sub interrupt 35: 80020065
[x]   Sub interrupt 36: 80020065
[x]   Sub interrupt 37: 80020065
[x]   Sub interrupt 38: 80020065
[x]   Sub interrupt 39: 80020065
[x]   Sub interrupt 40: 80020065
[x]   Sub interrupt 41: 80020065
[x]   Sub interrupt 42: 80020065
[x]   Sub interrupt 43: 80020065
[x]   Sub interrupt 44: 80020065
[x]   Sub interrupt 45: 80020065
[x]   Sub interrupt 46: 80020065
[x]   Sub interrupt 47: 80020065
[x]   Sub interrupt 48: 80020065
[x]   Sub interrupt 49: 80020065
[x]   Sub interrupt 50: 80020065
[x]   Sub interrupt 51: 80020065
[x]   Sub interrupt 52: 80020065
[x]   Sub interrupt 53: 80020065
[x]   Sub interrupt 54: 80020065
[x]   Sub interrupt 55: 80020065
[x]   Sub interrupt 56: 80020065
[x]   Sub interrupt 57: 80020065
[x]   Sub interrupt 58: 80020065
[x]   Sub interrupt 59: 80020065
[x]   Sub interrupt 60: 80020065
[x]   Sub interrupt 61: 80020065
[x]   Sub interrupt 62: 80020065
[x]   Sub interrupt 63: 80020065
[x]   Sub interrupt 64: 80020065
[x]   Sub interrupt 65: 80020065
[x]   Sub interrupt 66: 80020065
[x]   Sub interrupt 67: 80020065
[x]   Sub interrupt 68: 80020065
[x]   Sub interrupt 69: 80020065

[x] sceKernelReleaseSubIntrHandler handlers:
[x]   NULL handler: 80020068
[x]   Twice NULL: 80020068
[x]   Not NULL: 00000000
[x]   Twice not NULL: 80020068
